artemis/src/main/java/tech/pegasys/artemis/cli/BeaconNodeCommand.java

```diff
@@ -0,0 +1,491 @@
+      arity = "1")
+  private String dataStorageMode = DEFAULT_DATA_STORAGE_MODE;
+
+  // Beacon REST API
+
+  @Option(
+      names = {REST_API_PORT_OPTION_NAME},
+      paramLabel = "<INTEGER>",
+      description = "Port number of Beacon Rest API",
+      arity = "1")
+  private int restApiPort = DEFAULT_REST_API_PORT;
+
+  @Option(
+      names = {REST_API_DOCS_ENABLED_OPTION_NAME},
+      paramLabel = "<BOOLEAN>",
+      description = "Enable swagger-docs and swagger-ui endpoints",
+      arity = "1")
+  private boolean restApiDocsEnabled = DEFAULT_REST_API_DOCS_ENABLED;
+
+  @Option(
+      names = {REST_API_ENABLED_OPTION_NAME},
+      paramLabel = "<BOOLEAN>",
+      description = "Enables Beacon Rest API",
+      arity = "1")
+  private boolean restApiEnabled = DEFAULT_REST_API_ENABLED;
+
+  @Option(
+      names = {REST_API_INTERFACE_OPTION_NAME},
+      paramLabel = "<NETWORK>",
+      description = "Interface of Beacon Rest API",
+      arity = "1")
+  private String restApiInterface = DEFAULT_REST_API_INTERFACE;
+
+  private ArtemisConfiguration artemisConfiguration;
+  private BeaconNode node;
+
+  public String getConfigFile() {
+    return configFile;
+  }
+
+  public void parse(final String[] args) {
+    final CommandLine commandLine = new CommandLine(this).setCaseInsensitiveEnumValuesAllowed(true);
+    final EnvironmentVariableDefaultProvider environmentVariableDefaultProvider =
+        new EnvironmentVariableDefaultProvider(System.getenv());
+    final CommandLine.IDefaultValueProvider defaultValueProvider;
+    final Optional<File> maybeConfigFile = maybeFindConfigFile(commandLine, args);
+    if (maybeConfigFile.isPresent()) {
+      defaultValueProvider =
+          new CascadingDefaultProvider(
+              environmentVariableDefaultProvider,
+              new TomlConfigFileDefaultProvider(commandLine, maybeConfigFile.get()));
+    } else {
+      defaultValueProvider = environmentVariableDefaultProvider;
+    }
+    commandLine.setDefaultValueProvider(defaultValueProvider).execute(args);
+  }
+
+  @Override
+  public Integer call() {
+    try {
+      setLogLevels();
+      artemisConfiguration = artemisConfiguration();
+      node = new BeaconNode(artemisConfiguration);
+      node.start();
+      // Detect SIGTERM
+      Runtime.getRuntime()
+          .addShutdownHook(
+              new Thread(
+                  () -> {
+                    System.out.println("Teku is shutting down");
+                    node.stop();
+                  }));
+      return 0;
+    } catch (DatabaseStorageException ex) {
+      System.err.println(ex.getMessage());
+      System.exit(1);
+    } catch (Throwable t) {
+      System.err.println("Teku failed to start.");
+      t.printStackTrace();
+      System.exit(1);
+    }
+    return 1;
+  }
+
+  @VisibleForTesting
+  public ArtemisConfiguration getArtemisConfigurationDeprecated() {
```

Comment:
Looks like this method name is off.  Also wondering if we can make this package-private?  Might need to move the corresponding test:
```suggestion
  ArtemisConfiguration getArtemisConfiguration() {
```